framework/src/governance/lib/datazone/datazone-custom-asset-type-factory.ts

```diff
@@ -0,0 +1,189 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CustomResource, Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import { CfnProjectMembership } from 'aws-cdk-lib/aws-datazone';
+import { Effect, IRole, ManagedPolicy, PolicyDocument, PolicyStatement, Role, ServicePrincipal } from 'aws-cdk-lib/aws-iam';
+import { IFunction } from 'aws-cdk-lib/aws-lambda';
+import { ILogGroup } from 'aws-cdk-lib/aws-logs';
+import { Construct } from 'constructs';
+import { DataZoneCustomAssetTypeProps } from './datazone-custom-asset-type-props';
+import { DataZoneHelpers } from './datazone-helpers';
+import { Context, TrackedConstruct, TrackedConstructProps } from '../../../utils';
+import { DsfProvider } from '../../../utils/lib/dsf-provider';
+
+/**
+ * Properties for the DataZoneCustomAssetTypeFactory construct
+ */
+export interface DataZoneCustomAssetTypeFactoryProps {
+  /**
+   * The DataZone domain identifier
+   */
+  readonly domainId: string;
+  /**
+   * The removal policy for the custom resource
+   * @default RemovalPolicy.RETAIN
+   */
+  readonly removalPolicy?: RemovalPolicy;
+}
+
+/**
+ * Interface representing a DataZone custom asset type
+ */
+export interface CustomAssetType {
+  /**
+   * The domain identifier of the custom asset type
+   */
+  readonly domainId: string;
+  /**
+   * The name of the custom asset type
+   */
+  readonly name: string;
+  /**
+   * The project identifier owner of the custom asset type
+   */
+  readonly projectIdentifier: string;
+  /**
+   * The revision of the custom asset type
+   */
+  readonly revision: string;
+}
+
+/**
+ * Factory construct providing resources to create a DataZone custom asset type.
+ *
+ * @example
+ * new dsf.governance.DataZoneCustomAssetTypeFactory(this, 'CustomAssetTypeFactory', {
+ *  domainId: 'aba_dc999t9ime9sss',
+ * });
+ *
+ */
+export class DataZoneCustomAssetTypeFactory extends TrackedConstruct {
+  /**
+   * The CloudWatch Logs Log Group for the DataZone custom asset type creation
+   */
+  readonly createLogGroup: ILogGroup;
+  /**
+   * The Lambda Function for the DataZone custom asset type creation
+   */
+  readonly createFunction: IFunction;
+  /**
+   * The IAM Role for the DataZone custom asset type creation
+   */
+  readonly createRole: IRole;
+  /**
+   * The service token for the custom resource
+   */
+  readonly serviceToken: string;
+  /**
+   * The role used by the custom resource
+   */
+  readonly handlerRole: IRole;
```

Comment:
Do we support the bring you own role here? I do not recall I saw it in the props.

Reply:
duplicate with createRole which is now configurable